service/src/test/java/bio/terra/workspace/service/resource/referenced/ReferencedResourceServiceTest.java

```diff
@@ -320,7 +325,7 @@ void testEnumerate() {
       @BeforeEach
       void setup() throws Exception {
         // Make the Verify step always succeed
-        doReturn(true).when(mockCrlService).gcsBucketExists(any(), any());
+        doReturn(true).when(mockCrlService).canReadGcsBucket(any(), any());
```

Comment:
Are there existing tests that will hit the new `checkAccess` implementations instead of the mocks?

Reply:
I assumed so but this made me double check - we do have integration tests which run the new `checkAccess` as part of creating resources, but I realized we don't have any tests checking the "no access" path. Added one to exercise that as well for all three reference types